data_structures/src/chain.rs

```diff
@@ -2319,11 +2418,14 @@ impl TransactionsPool {
     /// ```
     pub fn vt_remove(&mut self, tx: &VTTransaction) -> Option<VTTransaction> {
         let key = tx.hash();
-        let transaction = self.vt_remove_inner(&key, true);
+        let transaction = self.vt_and_sh_remove_inner(&key, true);
 
         self.remove_inputs(&tx.body.inputs);
 
-        transaction
+        match transaction {
+            Some(Transaction::ValueTransfer(vt_tx)) => Some(vt_tx),
+            _ => None,
```

Comment:
Maybe panic instead of returning None?

Reply:
Nop, remove returns None when it doesnt find the key specified

Reply:
I see, I was thinking about the case when the transaction is `Some(Transaction::Script(_))`